juntagrico/admins/job_admin.py

```diff
@@ -26,47 +44,62 @@ class JobAdmin(PolymorphicInlineSupportMixin, RichTextAdmin):
     readonly_fields = ['free_slots']
 
     def has_change_permission(self, request, obj=None):
-        return (obj is None or obj.can_modify(request)) and super().has_change_permission(request, obj)
+        return (self.is_copy_view(request) or obj is None or obj.can_modify(request)
+                ) and super().has_change_permission(request, obj)
 
     def has_delete_permission(self, request, obj=None):
-        return (obj is None or obj.can_modify(request)) and super().has_delete_permission(request, obj)
+        return (self.is_copy_view(request) or obj is None or obj.can_modify(request)
+                ) and super().has_delete_permission(request, obj)
+
+    def get_readonly_fields(self, request, obj=None):
+        if self.is_copy_view(request):
+            return []  # special case for mass job copy action
+        return super().get_readonly_fields(request, obj)
+
+    def get_inlines(self, request, obj):
+        if self.is_copy_view(request):
+            return []  # special case for mass job copy action
+        return super().get_inlines(request, obj)
 
     @admin.action(description=_('Job mehrfach kopieren...'))
     @single_element_action('Genau 1 Job auswählen!')
     def mass_copy_job(self, request, queryset):
         inst, = queryset.all()
-        return HttpResponseRedirect('copy_job/%s/' % inst.id)
+        return HttpResponseRedirect(reverse('admin:action-mass-copy-job', args=[inst.id]))
 
     @admin.action(description=_('Jobs kopieren'))
     def copy_job(self, request, queryset):
         for inst in queryset.all():
-            newjob = RecuringJob(
-                type=inst.type, slots=inst.slots, time=inst.time)
+            time = inst.time
+            if not can_edit_past_jobs(request) and time <= timezone.now():
+                # create copy in future if job is in past and member can't edit past jobs
+                time = datetime.combine(timezone.now().date() + timedelta(7), time)
+            newjob = RecuringJob(type=inst.type, slots=inst.slots, time=time)
             newjob.save()
 
     def get_form(self, request, obj=None, **kwds):
-        if 'copy_job' in request.path:
-            return JobCopyForm
+        if self.is_copy_view(request):
```

Comment:
one could store the results of is_copy_view (cp) and can_edit_past_jobs (cepj) in variables and make a python swith like statement
if cp and cepj
return job copyform
if cp and not cepj
return Jobcopy to futureform
if not cp and cepy
return super.....getform...
if not cp and not cepj
return jobfutureadmin form

this will do the same thing but might make it more readable for dev which are not so fluent in logic

Reply:
I agree, that it is not easy to read. I'll add comments to help.